packages/openai/src/openai-chat-language-model.test.ts

```diff
@@ -144,6 +156,118 @@ describe('doGenerate', () => {
       (await server.getRequestHeaders()).get('Authorization'),
     ).toStrictEqual('Bearer test-api-key');
   });
+
+  it('should extract logprobs response', async () => {
+    const TEST_LOGPROBS = {
+      content: [
+        {
+          token: 'Hello',
+          logprob: -0.0009994634,
+          top_logprobs: [
+            {
+              token: 'Hello',
+              logprob: -0.0009994634,
+            },
+          ],
+        },
+        {
+          token: '!',
+          logprob: -0.13410144,
+          top_logprobs: [
+            {
+              token: '!',
+              logprob: -0.13410144,
+            },
+          ],
+        },
+        {
+          token: ' How',
+          logprob: -0.0009250381,
+          top_logprobs: [
+            {
+              token: ' How',
+              logprob: -0.0009250381,
+            },
+          ],
+        },
+        {
+          token: ' can',
+          logprob: -0.047709424,
+          top_logprobs: [
+            {
+              token: ' can',
+              logprob: -0.047709424,
+            },
+          ],
+        },
+        {
+          token: ' I',
+          logprob: -0.000009014684,
+          top_logprobs: [
+            {
+              token: ' I',
+              logprob: -0.000009014684,
+            },
+          ],
+        },
+        {
+          token: ' assist',
+          logprob: -0.009125131,
+          top_logprobs: [
+            {
+              token: ' assist',
+              logprob: -0.009125131,
+            },
+          ],
+        },
+        {
+          token: ' you',
+          logprob: -0.0000066306106,
+          top_logprobs: [
+            {
+              token: ' you',
+              logprob: -0.0000066306106,
+            },
+          ],
+        },
+        {
+          token: ' today',
+          logprob: -0.00011093382,
+          top_logprobs: [
+            {
+              token: ' today',
+              logprob: -0.00011093382,
+            },
+          ],
+        },
+        {
+          token: '?',
+          logprob: -0.00004596782,
+          top_logprobs: [
+            {
+              token: '?',
+              logprob: -0.00004596782,
+            },
+          ],
+        },
+      ],
+    };
+
+    prepareJsonResponse({ logprobs: TEST_LOGPROBS });
+
+    const provider = createOpenAI({ apiKey: 'test-api-key' });
+
+    const response = await provider
+      .chat('gpt-3.5-turbo', { logprobs: 1 })
+      .doGenerate({
+        inputFormat: 'prompt',
+        mode: { type: 'regular' },
+        prompt: TEST_PROMPT,
+      });
```

Comment:
an additional test for streaming would be good, since it's a separate implementation (same for the completion mode)